content/doc/book/managing/cli.adoc

```diff
@@ -0,0 +1,285 @@
+:sectanchors:
+:toc:
+:imagesdir: /doc/book/resources
+:hide-uri-scheme:
+
+= Jenkins CLI
+
+
+////
+Pages to mark as deprecated by this document:
+
+https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+CLI
+https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+SSH
+////
+
+
+Jenkins has a built-in command line interface that allows users and
+administrators to access Jenkins from a script or shell environment. This can
+be convenient for scripting of routine tasks, bulk updates, troubleshooting,
+and more.
+
+The command line interface can be accessed over SSH or with the Jenkins CLI
+client, a `.jar` file distributed with Jenkins. The SSH approach is preferred
+over the CLI client as it is considered more secure.
+
+////
+Background:
+23:19 <@rtyler> danielbeck: I'm writing CLI documentation, should I just encourage people to use SSH or stick with the default of jenkins-clir.jar?
+23:20 <@rtyler> we've told enough people to turn off the CLI over the past year...
+23:20 <@rtyler> :/
+23:21 <@danielbeck> it's probably better in general, but e.g. firewalls rules may require regular CLI use
+23:21 <@danielbeck> in any case, auth needs to use SSH keys
+////
+
+
+== Using the CLI
+
+By default Jenkins will boot with a randomly assigned SSH port, which
+administrators may choose to override in the <<system-configuration#ssh-server,
+Configure System>> page. In order to determine the randomly assigned SSH port,
+inspect the headers returned on a Jenkins URL, for example:
+
+[source]
+----
+% curl -Lv https://JENKINS_URL/login 2>&1 | grep 'X-SSH-Endpoint' 
+< X-SSH-Endpoint: localhost:53801
+%
+----
+
+With the random SSH port (`53801` in this example), and <<Authentication>>
+configured, any modern SSH client may securely execute CLI commands.
+
+
+In order to determine the available CLI commands, execute the `help` command,
+which will provide a list of commands available:
+
+[source]
+----
+% ssh -l kohsuke -p 53801 localhost help
+  add-job-to-view
+    Adds jobs to view.
+  build
+    Builds a job, and optionally waits until its completion.
+[...]
+%
+----
+
+=== Authentication
+
+Whichever user used for authentication with the Jenkins master must have the
+`Overall/Read` permission in order to _access_ the CLI. The user may require
+additional permissions depending on the commands executed.
+
+Whether using the CLI via SSH, or with the CLI client, both rely primarily on
+SSH-based public/private key authentication. In order to add an SSH public key
+for the appropriate user, navigate to
+`https://JENKINS_URL/user/USERNAME/cnofigure` and paste an SSH public key
+into the appropriate text area.
+
+image::managing/cli-adding-ssh-public-keys.png["Adding public SSH keys for a user", role=center]
+
+=== Common Commands
+
+Jenkins has a number of built-in CLI commands which can be found in every
+Jenkins environment, such as `build` or `list-jobs`. Plugins may also provide
+CLI commands however; in order to determine the full list of commands available
```

Comment:
I would strike 'however.' Also, we already talk about using the 'help' command above, in the Using the CLI section. Maybe we should strike that copy and keep this one, since this progresses into a further explanation of the common commands.